src/engine/enginemaster.cpp

```diff
@@ -443,48 +530,233 @@ void EngineMaster::process(const int iBufferSize) {
     }
 
     if (masterEnabled) {
-        // Mix the three channels together. We already mixed the busses together
-        // with the channel gains and overall master gain.
-        if (!m_pMasterTalkoverMix->toBool()) {
-            // Add Talkover to Master output
-            SampleUtil::copy4WithGain(m_pMaster,
-                    m_pOutputBusBuffers[EngineChannel::LEFT], 1.0,
-                    m_pOutputBusBuffers[EngineChannel::CENTER], 1.0,
-                    m_pOutputBusBuffers[EngineChannel::RIGHT], 1.0,
+        // Mix the crossfader orientation buffers together into the master mix
+        SampleUtil::copy3WithGain(m_pMaster,
+            m_pOutputBusBuffers[EngineChannel::LEFT], 1.0,
+            m_pOutputBusBuffers[EngineChannel::CENTER], 1.0,
+            m_pOutputBusBuffers[EngineChannel::RIGHT], 1.0,
+            iBufferSize);
+
+        // Process master, booth, and record/broadcast buffers according to the
+        // TalkoverMixMode configured in DlgPrefSound
+
+        // If a musician using a microphone plays on beat with the output they hear
+        // from Mixxx, when the input is processed by Mixxx and sent back out of Mixxx,
+        // they should hear themselves playing on beat. Due to the nature of digial audio,
+        // the user does not hear the output of Mixxx in the instant Mixxx is processing
+        // it. There is an output latency between the time Mixxx processes the audio
+        // and the user hears it. So if the microphone user plays on beat with what they
+        // hear, they will be playing out of sync with the engine's processing by the
+        // output latency. Additionally, Mixxx gets input signals delayed by the
+        // input latency. By the time Mixxx receives the input signal, a full
+        // round trip through the signal chain has elapsed since Mixxx processed the
+        // output signal. Therefore, to preserve the relative timing of the input with
+        // Mixxx's outputs (headphones, master, booth, and record/broadcast),
+        // the outputs must be delayed by the round trip latency before mixing
+        // the inputs into them. Unfortunately this adds some latency to Mixxx's outputs,
+        // so manipulating controls in Mixxx will feel less responsive. However,
```

Comment:
This is wrong, because the DJ must not hear the delayed signal. booth and headphones must not be artificial delayed.